### PR TITLE
add test for template wizard

### PR DESCRIPTION
**Pull request by gouyang**
Signed-off-by: Guohua Ouyang <gouyang@redhat.com>

### CODE REVIEW

File: frontend/integration-tests/tests/kubevirt/template.wizard.scenario.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  const leakedResources = new Set<string>();
+  const vmName = `vm-${testName}`;
+  const tmplName = `template-${testName}`;
+  const operatingSystem = 'Red Hat Enterprise Linux 7.6';
+  const flavor = 'small';
+  const workloadProfile = 'generic';
+  const sourceURL = 'https://download.cirros-cloud.net/0.4.0/cirros-0.4.0-x86_64-disk.img';
+  const sourceContainer = 'kubevirt/cirros-registry-disk-demo:latest';
+  const pxeInterface = 'nic1';
+  const provisionMethods = OrderedMap<string, (provisionSource: string) => void>()
```
Comment (rhrazdil) [inline frontend/integration-tests/tests/kubevirt/template.wizard.scenario.ts]: Seems we are reusing these. Consider moving these to mocks.ts.

File: frontend/integration-tests/tests/kubevirt/template.wizard.scenario.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    .set('URL', async function(provisionSource){
+      await vmView.provisionSourceButton.click();
+      await vmView.provisionSourceMenu.element(by.linkText(provisionSource)).click();
+
+      await vmView.provisionSourceURL.sendKeys(sourceURL);
+    });
+
+  async function fillBasicSettings(provisionMethod: (provisionSource: string) => void, provisionSourceName: string, template: boolean){
+    await browser.wait(until.presenceOf(vmView.nameInput), VM_WIZARD_LOAD_TIMEOUT);
+    if (template) {
```
Comment (rhrazdil) [inline frontend/integration-tests/tests/kubevirt/template.wizard.scenario.ts]: Why test for template when this scenario focuses on templates? IIUC you are creating template, not a VM from it, so there is no need to have non-template related stuff. Creating VM from template may be better to include in vm.wizard.scenario.ts. ATM however requires extending Wizard class.

File: frontend/integration-tests/tests/kubevirt/template.wizard.scenario.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+      // Confirm to create VM
+      await vmView.nextButton.click();
+      await browser.wait(until.not(until.textToBePresentInElement(vmView.wizardContent, 'Creation of VM in progress')), VM_WIZARD_LOAD_TIMEOUT);
+
+      // Check for error and close wizard
+      expect(errorMessage.isPresent()).toBe(false);
+      leakedResources.add(JSON.stringify({name: vmName, namespace: testName, kind: 'vm'}));
+      await browser.wait(until.elementToBeClickable(vmView.nextButton), PAGE_LOAD_TIMEOUT).then(() => vmView.nextButton.click());
+      // Verify VM is created and running
```
Comment (suomiy) [inline frontend/integration-tests/tests/kubevirt/template.wizard.scenario.ts]: @rhrazdil already mentioned, but there is no running VM from a template wizard

File: frontend/integration-tests/tests/kubevirt/template.wizard.scenario.ts [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    }
+
+    if (template) {
+      await vmView.startVMOnCreation.click();
+    }
+
+    await vmView.nextButton.click();
+  }
+
+  async function fillVMNetworking(provisionSourceName: string){
```
Comment (suomiy) [inline frontend/integration-tests/tests/kubevirt/template.wizard.scenario.ts]: nit: seems there is more stuff to be shared with vm.wizard.scenario

Comment (suomiy) [reply frontend/integration-tests/tests/kubevirt/template.wizard.scenario.ts]: sorry, I just discovered #212

File: frontend/integration-tests/tests/kubevirt/template.wizard.scenario.ts (hunk 1 above)
Comment (gouyang) [reply frontend/integration-tests/tests/kubevirt/template.wizard.scenario.ts]: > Hello, nice effort!
> Could you please update the scenario to reflect changes that are being introduced in #212 ?

OK, I will update it once 212 is merged.

**Comment by gouyang** [comment]
> Hello, nice effort!
> Could you please update the scenario to reflect changes that are being introduced in #212 ?

Okay, I will update this after #212 is merged.

**Comment by gouyang** [comment]
Will send a new PR instead of updating this as it need to re-write everything.

**Pull request closed without merging**